drivers/flash/flash_stm32_qspi.c

```diff
@@ -132,38 +136,55 @@ static inline void qspi_set_address_size(const struct device *dev,
 	cmd->AddressSize = QSPI_ADDRESS_24_BITS;
 }
 
-static inline void qspi_prepare_quad_read(const struct device *dev,
+static inline int qspi_prepare_quad_read(const struct device *dev,
 					  QSPI_CommandTypeDef *cmd)
 {
 	struct flash_stm32_qspi_data *dev_data = dev->data;
 
-	if (IS_ENABLED(STM32_QSPI_USE_QUAD_IO) && dev_data->flag_quad_io_en) {
+	if (dev_data->flag_quad_io_en) {
```

Comment:
```suggestion
	
	cmd->AddressMode = ((dev_data->mode == JESD216_MODE_114)
				? QSPI_ADDRESS_1_LINE
				: QSPI_ADDRESS_4_LINES);
```

Reply:
`switch` allows us to detect the case when some mode is not yet supported (i.e. `JESD216_MODE_444`) and return `-ENOTSUP` in such case. That eases figuring out why something is not working.

Alternatively, I can add `__ASSERT` then the ternary operator is ok.